pkg/cephmgr/osd/device.go

```diff
@@ -141,6 +141,15 @@ func partitionBluestoreDevice(context *clusterd.Context, config *osdConfig) erro
 		return fmt.Errorf("failed to save partition scheme. %+v", err)
 	}
 
+	// update the uuid of the disk in the inventory in memory
+	logger.Debugf("Updating disk uuid %s on device %s", scheme.DiskUUID, config.deviceName)
+	for _, disk := range context.Inventory.Local.Disks {
+		if disk.Name == config.deviceName {
+			logger.Debugf("Updated uuid on device %s", config.deviceName)
+			disk.UUID = scheme.DiskUUID
```

Comment:
so this is the meat of the change, and most everything else is the hardware inventory slices holding pointers to hardware structs now instead of the structs themselves?